### PR TITLE
AREA-41 - Create integration repository

### PR DESCRIPTION
**Pull request by Thomaltarix**
I created the integration repository.

It makes us able to fetch all integrations, or integrations by pages, or by pages and size.

>[!Note]
> Since I'm new with this technos and logics, please take a look at the logic of my work 👍

### CODE REVIEW

File: mobile/lib/repositories/integration.repository.dart [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import 'package:http/http.dart' as http;
+import 'package:triggo/api/call.dart';
+import 'package:triggo/api/response.dart';
+import 'package:triggo/repositories/credentials.repository.dart';
+import 'package:triggo/repositories/dtos/integration.dtos.dart';
+
+class IntegrationRepository {
+  final http.Client? client;
+  final CredentialsRepository? credentialsRepository;
```
Comment (Yann-Masson) [inline mobile/lib/repositories/integration.repository.dart:9]: Let's say that you MUST give a credential repository
```suggestion
  final CredentialsRepository credentialsRepository;
```

File: mobile/lib/repositories/integration.repository.dart [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  Future<Response<OutGetIntegrationDTO>> getIntegrationByPage(int page) async {
+    final response = await call(
+      method: 'GET',
+      endpoint: '/user/integrations/?page=$page',
+      headers: {
+        'Authorization':
+            'Bearer ${await credentialsRepository!.getAccessToken()}'
+      },
+      client: client,
+    );
```
Comment (Yann-Masson) [inline mobile/lib/repositories/integration.repository.dart:44]: I don't refuse this PR for that but it's just my opinion
```suggestion
    final accessToken = await credentialsRepository.getAccessToken();
    final response = await call(
      method: 'GET',
      endpoint: '/user/integrations/?page=$page',
      headers: {
        'Authorization':
            'Bearer ${accessToken}'
      },
      client: client,
    );
```

File: mobile/lib/repositories/integration.repository.dart [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  Future<Response<OutGetIntegrationDTO>> getIntegrations() async {
+    final response = await call(
+      method: 'GET',
+      endpoint: '/user/integrations',
+      headers: {
+        'Authorization':
+            'Bearer ${await credentialsRepository!.getAccessToken()}'
+      },
+      client: client,
+    );
```
Comment (Yann-Masson) [inline mobile/lib/repositories/integration.repository.dart:22]:
```suggestion
    final accessToken = await credentialsRepository.getAccessToken();
    final response = await call(
      method: 'GET',
      endpoint: '/user/integrations',
      headers: {
        'Authorization':
            'Bearer ${accessToken}'
      },
      client: client,
    );
```

**Review by Yann-Masson — CHANGES_REQUESTED** [review]
Nice work!
I love those tests! ❤️

File: mobile/lib/models/integration.model.dart [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
+class Integration {
   final String name;
   String description = '';
   String imageUrl = '';
   String iconUrl = '';
   int actionNumber = 0;
   int reactionNumber = 0;
 
-  Service({required this.name});
+  Integration({required this.name});
```
[the remark this answers is not in the document]
Comment (Thomaltarix) [reply mobile/lib/models/integration.model.dart:9]: You're right, I think I misunderstood the usage of the models when I did it

**Review by Yann-Masson — DISMISSED** [review]

**Review by Yann-Masson — APPROVED** [review]

**Pull request merged**